src/guides/legend.jl

```diff
@@ -1,4 +1,11 @@
-legend!(fg::FigureGrid; kwargs...) = legend!(fg.figure[:, end+1], fg; kwargs...)
+function legend!(fg::FigureGrid; kwargs...)
+ 
+    position, attr = legend_attributes(kwargs, false)
```

Comment:
For the `titleposition`, I have a strong preference that a horizontal legend should have its title on the left. This also matches ggplot2.

![image](https://user-images.githubusercontent.com/6280307/136937763-d1a0821a-4c7f-4c23-a777-6540671bbcf9.png)

The `nbanks` is here because I would like to combine colorbars and legend like

![](https://user-images.githubusercontent.com/6280307/91709239-c2851980-eb82-11ea-9ae4-6b232ad2c54b.png)

(nbanks should be 2 if the colorbar is horizontal.)

I played around with this in `TabularMakie.jl` but I didn't manage to get it to a satisfactory state.